site/src/pages/UserSettingsPage/SchedulePage/SchedulePage.tsx

```diff
@@ -0,0 +1,62 @@
+import { FC } from "react";
+import { ErrorAlert } from "components/Alert/ErrorAlert";
+import { Section } from "components/SettingsLayout/Section";
+import { ScheduleForm } from "./ScheduleForm";
+import { useMe } from "hooks/useMe";
+import { Loader } from "components/Loader/Loader";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import {
+  updateUserQuietHoursSchedule,
+  userQuietHoursSchedule,
+} from "api/queries/settings";
+import { displaySuccess } from "components/GlobalSnackbar/utils";
+
+export const SchedulePage: FC = () => {
+  const me = useMe();
+  const queryClient = useQueryClient();
+
+  const {
+    data: quietHoursSchedule,
+    error,
+    isLoading,
+    isError,
+  } = useQuery(userQuietHoursSchedule(me.id));
+
+  const updateSchedule = updateUserQuietHoursSchedule(me.id, queryClient);
+  const {
+    mutate: onSubmit,
+    error: mutationError,
+    isLoading: mutationLoading,
+  } = useMutation({
+    ...updateSchedule,
+    onSuccess: async () => {
+      await updateSchedule.onSuccess();
+      displaySuccess("Schedule updated successfully");
+    },
+  });
```

Comment:
Instead of initializing the options and overriding the `onSuccess` calling the parent why not do something like:
```ts
const {
  mutateAsync: onSubmit,
  error: mutationError,
  isLoading: mutationLoading,
} = useMutation(updateUserQuietHoursSchedule(me.id, queryClient));

// In the form props
<ScheduleForm
  onSubmit={ async () => {
    await onSubmit()
    displaySuccess("Schedule updated successfully");
  }}
/> 
```